Automattic/vip-cli
# Environment variable CLI commands (list, set, delete)

## Description

Adding commands to list, set, and delete environment variables. Welcoming feedback to help finalize the UX of these commands ~while I work on tests, error handling, and event tracking~.

## List

```
> vip @XXXX config envvar list
For security, the values of environment variables cannot be retrieved.

┌─────────────────┬────────────┐
│ name            │ value      │
├─────────────────┼────────────┤
│ CUSTOM_ENV_VAR  │ ********** │
├─────────────────┼────────────┤
│ CUSTOM_ENV_VAR2 │ ********** │
└─────────────────┴────────────┘
```

It also accepts a `--format` option:

```
> vip @XXXX config envvar list --format=csv
"name","value"
"CUSTOM_ENV_VAR","**********"
"CUSTOM_ENV_VAR2","**********"
```

## Set

This command will prompt for the value to avoid sensitive data in the user's shell history. 

```
> vip @XXXX config envvar set MY_VARIABLE
For multiline input, use the --from-file option.

✔ Enter the value for MY_VARIABLE: · testing
✔ Please confirm the input value above (y/N) · true
Successfully set environment variable "MY_VARIABLE"
```

Alternatively, a `--from-file` option lets users load the value from a file. I attempted `stdin` support, but I couldn't prevent `enquirer` from treating it as the user's response to the confirmation prompt. 

```
> vip @XXXX config envvar set MY_VARIABLE --from-file=my-variable-value.txt
===== Received value printed below =====
line 1
line 2
line 3
===== Received value printed above =====

✔ Please confirm the input value above (y/N) · true
Successfully set environment variable "MY_VARIABLE"
```

There is a `--skip-confirmation` option.

## Delete

```
> vip @XXXX config envvar delete MY_VARIABLE
✔ Type MY_VARIABLE to confirm deletion: · MY_VARIABLE
✔ Are you sure? Deletion is permanent (y/N) · true
Successfully deleted environment variable "MY_VARIABLE"
```

This also provides a `--skip-confirmation` option.

## Steps to Test

Example:

1. Check out PR.
1. Run `npm run build`
2. Run `npm link`
1. Run node `./dist/bin/vip config`
